.github/workflows/build.yml

```diff
@@ -4,41 +4,150 @@ on:
   push:
     branches: main
   pull_request:
-    branches: "*"
+    branches: '*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: ["3.10"]
+    runs-on: ubuntu-latest
+
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Base Setup
+      uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+    - name: Install dependencies
+      run: python -m pip install -U "jupyterlab>=4.0.0,<5"
+
+    - name: Lint the extension
+      run: |
+        set -eux
+        jlpm
+        jlpm run lint:check
+
+    - name: Test the extension
+      run: |
+        set -eux
+        jlpm run test
```

Comment:
The repo uses `codecov` for testing the extension and this workflow was previously installing the dependency and then for running the tests in this portion, it used `pytest`, like so: https://github.com/jupyter-robotics/ipylgbst/blob/4963d475a269032e1ece97e47413c3659c0686ed/.github/workflows/build.yml#L23-L31 I think this part of the initial workflow needs to be adjusted, as it's probably the cause of the `build` workflow failing.